File: mobile/lib/input/dashboard_input.dart
enum ControlId {
  parkingBrake('parking_brake'),
  turnSignalLeft('turn_signal_left'),
  turnSignalRight('turn_signal_right'),
  headlightToggle('headlight_toggle'),
  highBeamToggle('high_beam_toggle'),
  wipers('wipers'),
  cruiseToggle('cruise_toggle'),
  cruiseSetResume('cruise_set_resume'),
  engineStart('engine_start');

  const ControlId(this.wireValue);

  /// Snake_case value used on the wire, matching protocol/schema/controls.json.
  final String wireValue;
}

enum ActionType {
  toggle('toggle'),
  press('press'),
  release('release'),
  holdConfirm('hold_confirm');

  const ActionType(this.wireValue);

  /// Snake_case value used on the wire, matching protocol/schema/controls.json.
  final String wireValue;
}

/// Receives discrete dashboard control events from the UI.
class DashboardInput {
  void Function(ControlId control, ActionType action)? _onControlActivated;

  /// Registers the callback that receives dashboard control events.
  void onControlActivated(
    void Function(ControlId control, ActionType action) callback,
  ) {
    _onControlActivated = callback;
  }

  /// Reports a dashboard control event.
  void activate(ControlId control, ActionType action) {
    _onControlActivated?.call(control, action);
  }
}

File: android/app/src/main/kotlin/dev/fazrigading/wheeldeck/data/services/DashboardInput.kt
package dev.fazrigading.wheeldeck.data.services

import dev.fazrigading.wheeldeck.domain.models.ActionType
import dev.fazrigading.wheeldeck.domain.models.ControlId

/// Receives discrete dashboard control events from the UI.
class DashboardInput {
    private var onControlActivated: ((ControlId, ActionType) -> Unit)? = null

    /// Registers the callback that receives dashboard control events.
    fun onControlActivated(callback: (ControlId, ActionType) -> Unit) {
        onControlActivated = callback
    }

    /// Reports a dashboard control event.
    fun activate(control: ControlId, action: ActionType) {
        onControlActivated?.invoke(control, action)
    }
}
